### PR TITLE
Adding support for identity providers export

### PR DESCRIPTION
**Pull request by kazvictor**
fixes ca-api-gateway/gateway-developer-plugin#39

### CODE REVIEW

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        ObjectWriter yamlWriter = jsonTools.getObjectWriter(JsonTools.YAML);
+        try (OutputStream fileStream = Files.newOutputStream(servicesFile.toPath())) {
+            yamlWriter.writeValue(fileStream, identityProviderBeans);
+        } catch (IOException e) {
+            throw new WriteException("Exception writing identity providers config file", e);
+        }
+    }
+
+    private IdentityProvider getIdentityProviderBean(IdentityProviderEntity identityProviderEntity) {
```
Comment (kazvictor) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java]: Code Smell Code Smell: Remove this unused method parameter "identityProviderEntity". ([squid:S1172](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=18&issues=AWVYvP782xJdnw6Qj7pT&open=AWVYvP782xJdnw6Qj7pT)

[ISSUE_KEY]: AWVYvP782xJdnw6Qj7pT

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/IdentityProviderLinker.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayexport.tasks.explode.linker;
+
+import com.ca.apim.gateway.cagatewayexport.tasks.explode.bundle.Bundle;
+
+public class IdentityProviderLinker implements EntitiesLinker {
+    @Override
+    public void link(Bundle filteredBundle, Bundle bundle) {
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/IdentityProviderLinker.java]: Code Smell Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1186))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=18&issues=AWV8X8eCX2sY87p0ktEp&open=AWV8X8eCX2sY87p0ktEp)

[ISSUE_KEY]: AWV8X8eCX2sY87p0ktEp

Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/IdentityProviderLinker.java]: Code Smell Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1186))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=18&issues=AWV8X8eCX2sY87p0ktEp&open=AWV8X8eCX2sY87p0ktEp)

[ISSUE_KEY]: AWV8X8eCX2sY87p0ktEp

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/bundle/entity/IdentityProviderEntity.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return extensionXml;
+    }
+
+    public static class Builder {
+        private String name;
+        private String id;
+        private IdentityProviderType idProviderType;
+        private Map<String, Object> properties;
+        private Element extensionXml;
+
```
Comment (joao-borges) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/bundle/entity/IdentityProviderEntity.java]: I think you may remove the getters, they are not used.

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    private final JsonTools jsonTools;
+
+    public IdentityProviderWriter(DocumentFileUtils documentFileUtils, JsonTools jsonTools) {
+        this.documentFileUtils = documentFileUtils;
+        this.jsonTools = jsonTools;
+    }
+
+    @Override
+    public void write(Bundle bundle, File rootFolder) {
+        File configFolder = new File(rootFolder, "config");
```
Comment (joao-borges) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java]: look into ListenPortWriter#write, I made a utility method that does all of this, You may want to use it :)

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/bundle/entity/IdentityProviderEntity.java (hunk 3 above)
Comment (yves-chan) [reply src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/bundle/entity/IdentityProviderEntity.java]: True, I will remove unused methods

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java (hunk 4 above)
Comment (yves-chan) [reply src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/IdentityProviderWriter.java]: Will use the writehelper method!

**Pull request merged**